src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -53,39 +54,45 @@ public boolean validate() {
         final String regex = this.esSettings.get(ConfigConstants.OPENDISTRO_SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, null);
 
         if ((request.method() == RestRequest.Method.PUT || request.method() == RestRequest.Method.PATCH)
-                && regex != null
-                && !regex.isEmpty()
                 && this.content != null
                 && this.content.length() > 1) {
             try {
                 final Map<String, Object> contentAsMap = XContentHelper.convertToMap(this.content, false, XContentType.JSON).v2();
                 if (contentAsMap != null && contentAsMap.containsKey("password")) {
                     final String password = (String) contentAsMap.get("password");
 
-                    // Password can be null/empty for an existing user. Regex will validate password if present
-                    if (password != null && !password.isEmpty() && !regex.isEmpty() && !Pattern.compile("^"+regex+"$").matcher(password).matches()) {
-                        if(log.isDebugEnabled()) {
-                            log.debug("Regex does not match password");
-                        }
+                    // Password is not allowed to be empty if present.
+                    if (password != null && password.isEmpty()) {
```

Comment:
Under what condition `password` will be `null` here? It is executed with `contentAsMap.containsKey("password")` condition.

Reply:
In theory, contentAsMap.containsKey() prevents null password. But before checking isEmpty(), it's good to ensure the object "password" exists, isn't it?

Reply:
I don't see why two checks are necessary. The check `if (contentAsMap != null && contentAsMap.containsKey("password"))` looks redundant to me.

Reply:
If `contentAsMap` is `null` or `password` field doesn't exist, there's no need to validate password at all.

Reply:
How `contentAsMap` can be `null`? Does not check `password != null` supersede `contentAsMap.containsKey("password")`?

Reply:
I removed `password != null`.

Reply:
Will it be better to remove `contentAsMap != null && contentAsMap.containsKey("password")`?

Reply:
If removing `contentAsMap != null && contentAsMap.containsKey("password")`, we'd need to keep `if (password != null && password.isEmpty()) {` at [Line 65](https://github.com/opendistro-for-elasticsearch/security/blob/1bbf8e2f6d81c5fd7f28a861ba2d56785c630405/src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java#L65) and add `password != null` [Line 89](https://github.com/opendistro-for-elasticsearch/security/blob/1bbf8e2f6d81c5fd7f28a861ba2d56785c630405/src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java#L89). Will this look better? If so, I'll update it.

Reply:
Replace `contentAsMap != null && contentAsMap.containsKey("password")` with 
```
String password = contentAsMap.get("password");
if (password != null) {
```

Reply:
Done.